### ISSUE TITLE
test_http1_keep_alive_pre_request failure

### ISSUE THREAD

**Issue by Alessandro-Barbieri** [pgjones/hypercorn#37]
I've packaged hypercorn for the gentoo overlay guru and it fail at this test
Log here: https://784044.bugs.gentoo.org/attachment.cgi?id=700704
https://bugs.gentoo.org/784044

```
=================================== FAILURES ===================================
______________________ test_http1_keep_alive_pre_request _______________________

value = <trio.Nursery object at 0x7f78311575d0>

    async def yield_(value=None):
>       return await _yield_(value)

value      = <trio.Nursery object at 0x7f78311575d0>

/usr/lib/python3.7/site-packages/async_generator/_impl.py:106: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3.7/site-packages/async_generator/_impl.py:99: in _yield_
    return (yield _wrap(value))
        value      = <trio.Nursery object at 0x7f78311575d0>
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

client_stream = StapledStream(send_stream=<trio.testing.MemorySendStream object at 0x7f78311570d0>, receive_stream=<trio.testing.MemoryReceiveStream object at 0x7f7831157bd0>)

    @pytest.mark.trio
    async def test_http1_keep_alive_pre_request(
        client_stream: trio.testing._memory_streams.MemorySendStream,
    ) -> None:
        await client_stream.send_all(b"GET")
        await trio.sleep(2 * KEEP_ALIVE_TIMEOUT)
        # Only way to confirm closure is to invoke an error
        with pytest.raises(trio.BrokenResourceError):
>           await client_stream.send_all(b"a")
E           Failed: DID NOT RAISE <class 'trio.BrokenResourceError'>

client_stream = StapledStream(send_stream=<trio.testing.MemorySendStream object at 0x7f78311570d0>, receive_stream=<trio.testing.MemoryReceiveStream object at 0x7f7831157bd0>)

tests/trio/test_keep_alive.py:59: Failed
=============================== warnings summary ===============================
tests/asyncio/test_tcp_server.py::test_complets_on_half_close
  /usr/lib/python3.7/asyncio/base_events.py:612: RuntimeWarning: coroutine '_call_later' was never awaited
    self._ready.clear()

tests/middleware/test_dispatcher.py::test_trio_dispatcher_lifespan
  <string>:2: RuntimeWarning: coroutine '_call_later' was never awaited

tests/protocol/test_h2.py::test_protocol_handle_protocol_error
  /usr/lib/python3.7/site-packages/_pytest/runner.py:106: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    item.funcargs = None

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=========================== short test summary info ============================
FAILED tests/trio/test_keep_alive.py::test_http1_keep_alive_pre_request - Fai...
```

**Comment by pgjones** [comment]
Does this still error for you? I can't reproduce this.

**Comment by pgjones** [comment]
I think this is fixed in 0.13.0, please reopen if not
